### PR TITLE
Change Show more/less

### LINKED ISSUE #2150: Show more should change to Show less on clicking
**We are preparing to participate in Google Code-in, and have reserved this issue for participants in GCI - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more.**
### Please describe the problem (or idea)
When clicking on the Show More button it does not change to Show less
image
image
> What happened just before the problem occurred? Or what problem could this idea solve?
The text should change on clicking the button

### Please show us where to look
https://publiclab.org/wiki/timelapse

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by kevinzluo**
Fixes #2150  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [X] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [X] code is in uniquely-named feature branch and has no merge conflicts
* [X] PR is descriptively titled
* [X] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

I have set Show More to change to Show Less on clicking:
showmoreless
Please let me know if you would like me to change anything.

Thank you,
Kevin Luo

### CODE REVIEW

**Comment by gauravano** [comment]
Some of the tests are failing - 
test_error

You can also check by clicking on the Details link after the test name. Thanks!

**Comment by gauravano** [comment]
Hmm, I haven't gone through your code till now, but will do that after some hours. Although, from GIF, the fix looks good. We will fix the tests, shortly. 

Quick question: When it says "Show [num] more", this `num` is all the remaining entries left to be displayed?

**Comment by kevinzluo** [comment]
Yes that is all the remaining entries left.

**Comment by gauravano** [comment]
How many tasks are visible to you on GCI dashboard ?

**Comment by kevinzluo** [comment]
There are only 13 available tasks currently.
On my own dashboard I have only this one that I am working on.

**Comment by kevinzluo** [comment]
I have fixed it so that it works around the tests.

**Comment by kevinzluo** [comment]
@SidharthBansal @gauravano 
Could you please review this PR and approve the task on GCI?

Thank you.

**Comment by gauravano** [comment]
Hi @kevinzluo I was testing your PR on my system and found that there is some issue of ordering at each toggle. I am testing with notes. 

Notes_toggling
gi

File: app/views/grids/_notes.html.erb [hunk 1]
```diff
@@ -74,9 +74,21 @@
...
+    var sName = cName + " tr.extra";
+    
+    $(sName).toggleClass("hide");
+    if($(sName)[0].classList.contains("hide"))
+        var hiding = true;    
+    var myElement = event.target;    
+    if(event.target.classList.contains("caret")) 
+        myElement = event.target.parentElement;
+    if(!hiding) 
+        myElement.innerHTML = 'Show <%= nodes.length - 10 %> less <b class="caret" style = "transform: rotate(180deg);"></b>';
```
Comment (gauravano) [inline app/views/grids/_notes.html.erb]: I think it's fine just to use "Show more" and "Show less".

Comment (gauravano) [reply app/views/grids/_notes.html.erb]: Also, I like that you thought about the caret icon too. One more thing could be done changing the mouse cursor to hand on hovering 'show more' link.

**Comment by gauravano** [comment]
Also, remove this line https://github.com/publiclab/plots2/pull/3980/files#diff-779e0fb416873b1a765a452c7ddc5941R96 , which will stop the reordering of notes at each toggle. Thanks!

**Comment by kevinzluo** [comment]
@gauravano Thank you for your guidance!

I have set the cursor to change to hand and removed the number shown. I also set a minimum width because the columns kept shifting.

I commented out the setupGridSorters line because I felt we may need to re add it. Should I delete it completely?

**Comment by gauravano** [comment]
You can leave the line there but add to comment the purpose to line and it's effect in a short

**Comment by kevinzluo** [comment]
Got it. Thank you.

**Comment by gauravano** [comment]
@jywarren this one is complete, please have a look

**Comment by jywarren** [comment]
So awesome!!! Great teamwork too!!!

**Comment by publiclab-mimi** [comment]
Hello team!

I'm thinking that with Bootstrap 4 we may have a bug with this feature again. I've noticed that our grids have gotten really long, and I don't think that anything is being hidden anymore. 

Here's what I'm seeing in the text version:

Show More:Less

And here is what I'm seeing in the version with photos, where "Show More" is just text and no longer a button:

Show More:Less 2

**Pull request merged**